apps/src/templates/teacherDashboard/TeacherDashboard.jsx

```diff
@@ -19,142 +19,146 @@ import StandardsReport from '../sectionProgress/standards/StandardsReport';
 import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
 import i18n from '@cdo/locale';
 
-class TeacherDashboard extends Component {
-  static propTypes = {
-    studioUrlPrefix: PropTypes.string.isRequired,
-    sectionId: PropTypes.number.isRequired,
-    sectionName: PropTypes.string.isRequired,
-    studentCount: PropTypes.number.isRequired,
-    coursesWithProgress: PropTypes.array.isRequired,
+function TeacherDashboard({
+  studioUrlPrefix,
+  sectionId,
+  sectionName,
+  studentCount,
+  coursesWithProgress,
+  location
+}) {
+  const usePrevious = value => {
+    const ref = React.useRef();
+    React.useEffect(() => {
+      ref.current = value;
+    });
 
-    // Provided by React router in parent.
-    location: PropTypes.object.isRequired
+    return ref.current;
   };
 
-  componentDidUpdate(prevProps, prevState, snapshot) {
-    const previousTab = _.last(_.split(prevProps.location.pathname, '/'));
-    const newTab = _.last(_.split(this.props.location.pathname, '/'));
+  const useLocationChange = action => {
+    const prevLocation = usePrevious(location);
+    useEffect(() => {
+      if (prevLocation) {
```

Comment:
this checks first to make sure there is a previousLocation (otherwise it would send an event on the first page load)